### PR TITLE
Add a `Node.Data()` method to get a `Node`'s parent, key and value

### PR DESCRIPTION
**Pull request by na--**
This will allow external users to iterate over the values of a Node's path efficiently. Not sure about the name though, open for any better suggestions?

### CODE REVIEW

File: atlas.go [hunk 1]
```diff
@@ -35,6 +35,11 @@ func (n *Node) IsRoot() bool {
 	return n.root == n
 }
 
+// GetNodeData returns the previous node and the key and value of the current one.
+func (n *Node) GetNodeData() (*Node, string, string) {
```
Comment (mstoykov) [inline atlas.go:39]: I would argue this is a rare case in which named returns make sense ;) 

```suggestion
func (n *Node) GetNodeData() (prev *Node, key string, value string) {
```

I don't really have better name if we want all three at the same time. Separating themi in 2 methods can be

`Prev`/`Parent`
and 
`KeyValue`/`Data` !?!


I would Also prever to drop the `Get` from thsi method if possible ;)

Comment (na--) [reply atlas.go:39]: When iterating over them, users will probably almost always want all 3, and they can use `_` if they don't. For example, here's how encoding them in JSON worked out in k6:
```go
func (ts *TagSet) MarshalEasyJSON(w *jwriter.Writer) {
	w.RawByte('{')
	first := true

	n := (*atlas.Node)(ts)
	for !n.IsRoot() {
		prev, key, value := n.GetNodeData()
		if first {
			first = false
		} else {
			w.RawByte(',')
		}
		w.String(key)
		w.RawByte(':')
		w.String(value)
		n = prev
	}
	w.RawByte('}')
}
```

Which seems fairly readable? :+1: on removing the `Get` and the named returns though, so how about this:

```suggestion
func (n *Node) NodeData() (prev *Node, key string, value string) {
```

:question:

Comment (mstoykov) [reply atlas.go:39]: :+1: 

maybe add an example with this (simplified) either as Example or just as part of the docs :thinking:

Comment (na--) [reply atlas.go]: Resolved by the [force-push](https://github.com/mstoykov/atlas/compare/1ab3558876aaed3042eda88d0bc526d0c075027f..75bb75c1ca5bb17aaee0fa2abbd1aab9b1e74d10)

File: atlas_test.go [hunk 2]
```diff
@@ -151,6 +152,58 @@ func TestNodeIsRoot(t *testing.T) {
...
+		prev, key, value := iter.Data()
+		fmt.Printf("%s: %s\n", key, value)
+		iter = prev
+	}
+
+	// Output:
+	// bar: 2
+	// baz: 3
+	// foo: 1
+}
```
Comment (mstoykov) [inline atlas_test.go:171]: :heart:

**Review by mstoykov — APPROVED** [review]

**Pull request merged**